### PR TITLE
exception if --privacy-marker-transaction-signing-key-file not specified in non-zero-gas network

### PR DESCRIPTION
**Pull request by macfarla**
With privacy enabled, it is only valid to omit --privacy-marker-transaction-signing-key-file
if --min-gas-price=0
So - if --privacy-enabled and --min-gas-price > 0 and --privacy-marker-transaction-signing-key-file not specified, throw exception and prevent startup.

See #505

### CODE REVIEW

File: besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java [hunk 1]
```diff
@@ -909,7 +909,7 @@ void setBannedNodeIds(final List<String> values) {
   @Option(
       names = {"--target-gas-limit"},
       description =
-          "Sets target gas limit per block. If set each blocks gas limit will approach this setting over time if the current gas limit is different.")
+          "Sets target gas limit per block. If set each block's gas limit will approach this setting over time if the current gas limit is different.")
```
Comment (mark-terry) [inline besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java:912]: "set, each" ?

File: besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java [hunk 2]
```diff
@@ -1882,6 +1882,16 @@ private PrivacyParameters privacyParameters() {
...
             commandLine, "Please specify Enclave public key file path to enable privacy");
       }
+
+      if (Wei.ZERO.compareTo(minTransactionGasPrice) < 0) {
+        // if gas is required, cannot use random keys to sign private tx
+        // ie --privacy-marker-transaction-signing-key-file must be set
+        if (privacyMarkerTransactionSigningKeyPath == null) {
+          throw new ParameterException(
+              commandLine,
+              "Not a free gas network. --privacy-marker-transaction-signing-key-file must be specified and must be a funded account. Private transactions cannot be signed by random (non-funded) accounts in paid gas networks");
```
Comment (mark-terry) [inline besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java:1892]: "Private transactions cannot be signed by random (non-funded) accounts in paid gas networks"

Would you expect a user to connect that not specifying a key file will mean a random account is used? I'm not sure whether we can expect the user to have enough context to understand the last sentence.

File: besu/src/test/java/org/hyperledger/besu/cli/BesuCommandTest.java [hunk 3]
```diff
@@ -3073,6 +3089,15 @@ public void onchainPrivacyAndMultiTenancyCannotBeUsedTogether() {
         .startsWith("Privacy multi-tenancy and onchain privacy groups cannot be used together");
   }
 
+  @Test
+  public void privacyMarkerTransactionSigningKeyFileRequiredIfMinGasPriceNonZero() {
+    parseCommand("--privacy-enabled", "--privacy-public-key-file", ENCLAVE_PUBLIC_KEY_PATH);
```
Comment (mark-terry) [inline besu/src/test/java/org/hyperledger/besu/cli/BesuCommandTest.java:3094]: Does it make sense to rely on a default non-zero gas price, or would it be clearer to have an explicit non-zero gas price?

Comment (macfarla) [reply besu/src/test/java/org/hyperledger/besu/cli/BesuCommandTest.java:3094]: changing this is outside the scope of this PR

File: besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java (hunk 1 above)
Comment (macfarla) [reply besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java:912]: this sort of thing appears in the help both ways. If we're going to make it consistent that would be a bigger change than this PR.

File: besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java (hunk 2 above)
Comment (macfarla) [reply besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java:1892]: I got that from reading the docs. I think if you're using privacy you should read the docs. Also I asked @MadelineMurray to preapprove this error message

**Pull request merged**